Sources/NIO/EventLoopFuture.swift

```diff
@@ -217,6 +217,32 @@ public struct EventLoopPromise<Value> {
         self._resolve(value: result)
     }
 
+    /// Execute the given function and complete the promise.
+    ///
+    /// Specifically, the function is executed synchronously on the `EventLoop`
+    /// that was used to create the promise.
+    ///
+    /// - parameters:
+    ///     - body: The function to execute.
+    @inlinable
+    public func completeWithBlock(_ body: @escaping () throws -> Value) {
```

Comment:
So despite not being a fan of overloading, in this case I think we should probably overload the `completeWith` again for throwing functions. "Block" isn't really a word that's used in Swift.

Apart from the naming, the only thing I think we need here is a big, fat warning that you must not call into any blocking code here.

Reply:
Hah, that's the exact opposite code review feedback I gave. :wink:

Is there a compromise position we can get to here?

Reply:
@Lukasa haha oops. So if we find a good name, I'm (always) happy not to overload. But `completeWithClosure` doesn't sound great, `completeWithBlock` sounds like ObjC. But the real reason that tipped me into "let's overload here" is that the other two versions are spelled `completeWith` and are overloaded, so I thought let's follow prior art.

But really, I'm not crazy strong here just didn't like "block" (ObjC and may even be confused with "blocking") and then looked at the other two.

Reply:
Hmm, they are overloaded, but they're overloaded with non-confusing concrete types. Overloading with closures always makes me feel a bit more nervous. But ok, I'm fine with going back to `completeWith` I guess. Sorry @Hailong!

Reply:
@Lukasa I agree that closures are extra troublesome...

Reply:
`Result` uses `Result(catching: () throws -> T)` which is also weird ;)

Reply:
Maybe `completeWithClosure` isn't too bad? @Lukasa / @Hailong WDYT?

Reply:
WFM

Reply:
Thank you both! I have just changed it to `completeWithClosure`. I also reworded the doc a little bit and added the warning message, please let me know if anything is missing. 😄